### PR TITLE
[BPK-908] Fix lazy loading HOC not responding to div scroll events

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] For any new components I've made sure that the style can be extended by the consumer using a `className` override.
+ [x] For any new files I've included the [Apache 2.0 License header](https://github.com/Skyscanner/backpack/blob/master/packages/bpk-tokens/formatters/license-header.js#L2-L16) at the top of the file.
+ [x] I've updated `changelog.md` for any changes that need to be highlighted in the changelog.
+ [x] If I've updated `npm-shrinkwrap.json` those changes are intentional.

### CODE REVIEW

File: packages/bpk-component-image/src/withLazyLoading.js [hunk 1]
```diff
@@ -57,8 +58,10 @@ export default function withLazyLoading(Component, document) {
     }
 
     removeEventListeners() {
-      document.removeEventListener('scroll', this.checkInView);
+      const passiveArgs = this.supportsPassiveEvents() ? { passive: true } : {};
+      document.removeEventListener('scroll', this.checkInView, { capture: true, ...passiveArgs });
```
Comment (matthewdavidson) [inline packages/bpk-component-image/src/withLazyLoading.js]: When removing an event listener, do you have to provide the options? What happens if you omit them?

File: packages/bpk-mixins/src/mixins/_images.scss [hunk 2]
```diff
@@ -33,8 +33,14 @@
   position: relative;
   display: block;
   max-width: 100%;
+  transition: background-color $bpk-duration-base ease-in-out;
   background-color: $bpk-color-gray-100;
   overflow: hidden;
+  transition-delay: $bpk-duration-base;
+}
+
+@mixin bpk-image--show {
```
Comment (matthewdavidson) [inline packages/bpk-mixins/src/mixins/_images.scss]: This doesn't need to be a mixin - just have it inline. In fact the mixins for the image component are realistically never going to be used by consumers. We should remove them (perhaps outside the scope of this PR).

Comment (georgegillams) [reply packages/bpk-mixins/src/mixins/_images.scss]: Done here https://github.com/Skyscanner/backpack/pull/82

File: packages/bpk-component-image/src/withLazyLoading.js [hunk 3]
```diff
@@ -36,9 +38,11 @@ export default function withLazyLoading(Component, document) {
     }
 
     componentDidMount() {
-      const passiveArgs = this.supportsPassiveEvents() ? { passive: true } : false;
-      document.addEventListener('scroll', this.checkInView, passiveArgs);
+      const passiveArgs = this.supportsPassiveEvents() ? { passive: true } : {};
+      this.scrollEventArgs = { capture: true, ...passiveArgs };
```
Comment (matthewdavidson) [inline packages/bpk-component-image/src/withLazyLoading.js]: Hmm we should have a test for this behaviour.

I.e. a test that asserts that `this.scrollEventArgs` is the same after `componentDidMount ` and during `componentWillUnmount`.

Comment (georgegillams) [reply packages/bpk-component-image/src/withLazyLoading.js]: I feel like this is just testing javascript?? Maybe worthwhile in-case someone changes the args?

Comment (georgegillams) [reply packages/bpk-component-image/src/withLazyLoading.js]: Actually, come to think of it, we don't want to test that they are the same - as this is internal workings of the HOC. What we want to test is that the event listeners are, indeed, added and removed correctly

Comment (georgegillams) [reply packages/bpk-component-image/src/withLazyLoading.js]: IMHO

**Pull request merged**